lib/msf/core/exploit/remote/http/exchange.rb

```diff
@@ -0,0 +1,144 @@
+# -*- coding: binary -*-
+
+module Msf
+  class Exploit
+    class Remote
+      module HTTP
+        # This module provides a way of interacting with Exchange installations
+        module Exchange
+          include Msf::Exploit::Remote::HttpClient
+
+          # Taken from https://docs.microsoft.com/en-us/exchange/new-features/build-numbers-and-release-dates
+          # Full list of all known Exchange Server builds.
+          @@exchange_builds = []
+
+          raw_exchange_build_json = JSON.load(File.read(::File.join(Msf::Config.data_directory, 'exchange_versions.json')))
+          for server_version in raw_exchange_build_json['exchange_builds']
+            for build in server_version['builds']
+              @@exchange_builds << build
+            end
+          end
```

Comment:
Potential solution to lazy loading this:

```suggestion
       def exchange_builds
          return @exchange_builds if @exchange_builds
          @exchange_builds = []
          
          raw_exchange_build_json = JSON.load(File.read(::File.join(Msf::Config.data_directory, 'exchange_versions.json')))
          for server_version in raw_exchange_build_json['exchange_builds']
            for build in server_version['builds']
              @exchange_builds << build
            end
          end
      end
```

Reply:
Suggested solution implemented with https://github.com/rapid7/metasploit-framework/pull/16249/commits/6a8a2531bfb2a56e47ff260a26c033f162485ea4.